packages/plugin-npm-manifest/index.js

```diff
@@ -14,37 +14,32 @@ function linkDependency(blob, key, value) {
   const isValidSemver = isSemver(value);
   const regex = jsonRegExKeyValue(key, value);
 
-  return insertLink(blob.el, regex, {
-    pluginName: 'NpmManifest',
-    target: isValidSemver ? '$1' : '$2',
+  return insertLink(blob, regex, this, {
     type: isValidSemver ? 'liveResolverQuery' : 'git',
   });
 }
 
 function linkFile(blob, key, value) {
   const regex = jsonRegExValue(key, value);
-
-  return insertLink(blob.el, regex, {
-    pluginName: 'NpmManifest',
-    type: 'file',
-    path: blob.path,
-    target: '$1',
-  });
+  return insertLink(blob, regex, this, { type: 'file' });
 }
 
 export default {
   name: 'NpmManifest',
 
-  resolve({ target, path, type }) {
+  resolve(path, values, { type }) {
```

Comment:
The resolve function receives the RegExp result as the second argument in order to get rid of the captured group ($1, $2 ...) magic. In most of the cases you are interested the first value. In some rare cases like this npm plugin, you need to differentiate based on the value the next step e.g. calling resolver with different values.